### PR TITLE
statedb: updated storage/statedb/doc.go

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- Updated storage/statedb/doc.go

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: storage/statedb/doc.go [hunk 1]
```diff
@@ -18,5 +18,37 @@
...
+This prevents calling code from creating long chains of nodes that increase the access time.
+
+Database is an intermediate write layer between the trie data structures and
+the disk database. The aim is to accumulate trie writes in-memory and only
+periodically flush a couple tries to disk, garbage collecting the remainder.
+
+Source Files
+
+  - database.go    : Implementation of Database struct
+  - derive_sha.go  : Implementation of DeriveShaOrig used in go-ethereum
```
Comment (KimKyungup) [inline storage/statedb/doc.go]:
```suggestion
  - derive_sha.go  : Implementation of DeriveShaOrig used in Klaytn
```

File: storage/statedb/doc.go [hunk 2]
```diff
@@ -18,5 +18,37 @@
...
+  - encoding.go    : Implementation of 3 encodings: KEYBYTES, HEX and COMPACT
+  - errors.go      : Errors used in this package
+  - hasher.go      : Implementation of recursive and bottom-up hashing
+  - iterator.go    : Implementation of key-value trie iterator that traverses a Trie
+  - node.go        : Implementation of 4 types of nodes
```
Comment (KimKyungup) [inline storage/statedb/doc.go]:
```suggestion
  - node.go        : Implementation of nodes for Merkle Patricia Trie.
```

File: storage/statedb/doc.go (hunk 1 above)
Comment (ehnuje) [reply storage/statedb/doc.go]: Reflected! Thanks!

File: storage/statedb/doc.go (hunk 2 above)
Comment (ehnuje) [reply storage/statedb/doc.go]: Reflected! Thanks!

File: storage/statedb/doc.go [hunk 3]
```diff
@@ -18,5 +18,46 @@
...
 // Modified and improved for the klaytn development.
 
-// Package statdb implements the network stats reporting service.
+/*
+Package statedb implements the Merkle Patricia Trie structure used for state object trie.
+This package is used to read/write data from/to the state object trie.
+
+Overview of statedb package
+
+There are 3 key struct in this package: Trie, SecureTire and Database.
```
Comment (laggu) [inline storage/statedb/doc.go]: SecureTire -> SecureTrie :)

Comment (ehnuje) [reply storage/statedb/doc.go]: @laggu Hawk Eyes 👀

**Comment by ehnuje** [comment]
@KimKyungup @laggu Dear reviewers, could you please take a look into this change?

**Pull request merged**